### PR TITLE
495 default editorjs blocks

### PR DESCRIPTION
**Pull request by daphneslootmans**
## Type

- Enhancement

## Pull request description

Add editorjs plugins and additional custom blocks

### CODE REVIEW

File: src/Modules/Backend/assets/Backend/webpack/js/Components/TestComponent.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<script>
```
Comment (tijsverkoyen) [inline src/Modules/Backend/assets/Backend/webpack/js/Components/TestComponent.vue:1]: Is this test component needed?

File: src/Modules/Backend/assets/Backend/webpack/js/Backend.js [hunk 2]
```diff
@@ -50,7 +52,15 @@ export class Backend {
...
     this.session = new Session()
     this.ajaxContentEditable = new AjaxContentEditable(this.locale)
-    this.initToasts = new InitBsToasts()
+
+    // init media selector vue component
+    const vueApp = document.querySelector('[data-role="vue-app"]')
+    if (vueApp) {
+      const app = createApp()
+      // global component
+      app.component('TestComponent', TestComponent)
```
Comment (tijsverkoyen) [inline src/Modules/Backend/assets/Backend/webpack/js/Backend.js:61]: Is this test component needed? Or is this some debug-code for the media-library?

File: src/Modules/Backend/assets/Backend/webpack/js/Backend.js [hunk 3]
```diff
@@ -3,6 +3,8 @@
 import * as bootstrap from 'bootstrap/dist/js/bootstrap.js'
 import 'select2/dist/js/select2.full'
 import 'flatpickr'
+import { createApp } from 'vue'
+import TestComponent from './Components/TestComponent.vue'
```
Comment (tijsverkoyen) [inline src/Modules/Backend/assets/Backend/webpack/js/Backend.js:7]: Is this needed?

File: src/Modules/Backend/assets/Backend/webpack/scss/components/_codex-editor.scss [hunk 4]
```diff
@@ -31,19 +31,26 @@
 .ce-wrapper-label {
   color: $gray-700;
   margin-bottom: 0;
+  padding-right: 1rem;
```
Comment (tijsverkoyen) [inline src/Modules/Backend/assets/Backend/webpack/scss/components/_codex-editor.scss:34]: Isn't there a bootstrap variable that makes sense and can be used.

File: src/Modules/BlockEditor/assets/Backend/webpack/js/BlockEditor.js [hunk 5]
```diff
@@ -7,6 +7,13 @@ import Underline from '@editorjs/underline'
 import Button from './Blocks/Button'
 import Quote from './Blocks/Quote'
 import Raw from './Blocks/Raw'
+import TextImageBlock from './Blocks/TextImageBlock'
+import TextColumnBlock from './Blocks/TextColumnBlock'
+import { createApp } from 'vue'
+import TestComponent from '../../../../../Backend/assets/Backend/webpack/js/Components/TestComponent.vue'
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/BlockEditor.js:13]: Is this needed?

File: src/Modules/BlockEditor/assets/Backend/webpack/js/BlockEditor.js [hunk 6]
```diff
@@ -69,9 +92,21 @@ export class BlockEditor {
...
+        if (vueApp) {
+          const app = createApp()
+          // global component
+          app.component('TestComponent', TestComponent)
+          app.mount('.vue-app')
+        }
+      },
       onChange: () => {
         editor.save().then((outputData) => {
+          console.log('outputData: ', outputData)
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/BlockEditor.js:109]: Debug code?

File: src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextColumnBlock.js [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  constructor ({ data, block }) {
+    this.block = block
+    this.data = {
+      title1: data.title1,
+      text1: data.text1,
+      title2: data.title2,
+      text2: data.text2
+    }
+
+    console.log('data block: ', data)
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextColumnBlock.js:13]: Debug code?

File: src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextColumnBlock.js [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        center: 'ce-tune-alignment--center',
+        right: 'ce-tune-alignment--right'
+      }
+    }
+  }
+
+  render () {
+    // wrapper
+    const wrapper = this._make('div', this.CSS.wrapper)
+    // wrapper.classList.add(this.CSS.alignment[this.data.alignment])
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextColumnBlock.js:58]: Debug code?

File: src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextImageBlock.js [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+  renderSettings () {
+    return [
+      {
+        icon: '<svg xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path d="M288 64c0 17.7-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32H256c17.7 0 32 14.3 32 32zm0 256c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H256c17.7 0 32 14.3 32 32zM0 192c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 448c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"/></svg>',
+        label: 'Image left',
+        toggle: 'image', // <--- specify toggle group name
+        onActivate: () => {
+          this.data.imagePosition = 'left'
+          this.block.dispatchChange()
+          console.log(this.data.imagePosition)
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextImageBlock.js:98]: Debug code?

File: src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextImageBlock.js [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+        isActive: this.data.imagePosition === 'left'
+      },
+      {
+        icon: '<svg xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 448 512"><path d="M448 64c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32zm0 256c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32zM0 192c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 448c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"/></svg>',
+        label: 'Image right',
+        toggle: 'image', // <--- specify toggle group name
+        onActivate: () => {
+          this.data.imagePosition = 'right'
+          this.block.dispatchChange()
+          console.log(this.data.imagePosition)
```
Comment (tijsverkoyen) [inline src/Modules/BlockEditor/assets/Backend/webpack/js/Blocks/TextImageBlock.js:109]: Debug code?

**Comment by carakas** [comment]
Closing for now since we'll try a different editor

**Pull request closed without merging**